spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/client/loadbalancer/impl/scoped/CachedDiscoveryClient.java

```diff
@@ -0,0 +1,31 @@
+package org.springframework.cloud.client.loadbalancer.impl.scoped;
+
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.cloud.client.ServiceInstance;
+import org.springframework.cloud.client.discovery.DiscoveryClient;
+import org.springframework.core.env.Environment;
+
+import java.util.List;
+
+/**
+ * @author Spencer Gibb
+ */
+public class CachedDiscoveryClient implements ScopedDiscoveryClient {
+
+	private final DiscoveryClient delegate;
+	private final Environment environment;
+
+	public CachedDiscoveryClient(DiscoveryClient delegate, Environment environment) {
+		this.delegate = delegate;
+		this.environment = environment;
+	}
+
+	@Cacheable(cacheNames = "discovery-client", key = "${loadbalancer.client.name}")
```

Comment:
When will this cache expire?  I read that if the parameters change than the cache wont be returned but since parameters are not changing how will it be expired?